projects/igniteui-angular/src/lib/icon/icon.component.html

```diff
@@ -5,9 +5,13 @@
 </ng-template>
 
 <ng-template #svgImage>
-    <svg>
-        <use [attr.href]="getSvgKey"></use>
-    </svg>
+    <svg
+        viewBox="0 0 24 24"
+        preserveAspectRatio="xMidYMid meet"
+        xmlns="http://www.w3.org/2000/svg"
+        xmlns:xlink="http://www.w3.org/1999/xlink"
+        [innerHTML]="getSvg"
```

Comment:
I don't think that it's safe to assume that any SVG will have a 24x24 viewbox.
Can you use `[outerHTML]` instead?

Reply:
We can't bind to outerHTML in Angular. It throws an error.

Reply:
It does work for me: https://stackblitz.com/edit/angular-1kvoe2?file=src/app/app.component.html

In any case, I'm not able to understand the code fully.
As I see in the cache you are saving: `this._cachedSvgIcons.get(family).set(name, svg.outerHTML);`
And then here you create your own SVG and put the string from the Cache inside.
So you would get a duplicate `<svg>` tag? Or I'm missing something?

Reply:
Technically you can use outerHTML and it will work, however you will get the following error in the browser console:

```
ERROR DOMException: Failed to set the 'outerHTML' property on 'Element': This element has no parent node.
```

You can refer to [this issue](https://github.com/angular/angular/issues/41576).

No duplicate svg tags are being created. The icon service creates a div and places the content of the fetched string (svg) inside of it (`div.innerHTML = value`). Then we query the div for the svg element we just added as innerHTML to modify some of its attributes before caching it. Finally we cache the icon for the specified family.